### PR TITLE
Add missing swapchain image destroy

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

**Review by bradgrantham-lunarg — APPROVED** [review]

File: framework/encode/vulkan_state_tracker.cpp [hunk 1]
```diff
@@ -1935,6 +1935,7 @@ void VulkanStateTracker::DestroyState(vulkan_wrappers::SwapchainKHRWrapper* wrap
     std::unique_lock<std::mutex> lock(state_table_mutex_);
     for (auto entry : wrapper->child_images)
     {
+        DestroyState(entry);
```
Comment (mikes-lunarg) [inline framework/encode/vulkan_state_tracker.cpp:1938]: Does this need similar logic to https://github.com/LunarG/gfxreconstruct/blob/dev/framework/encode/vulkan_handle_wrapper_util.h#L650-L658 to handle child images that belong to multiple swapchains?

Comment (ziga-lunarg) [reply framework/encode/vulkan_state_tracker.cpp]: Good catch

Comment (mikes-lunarg) [reply framework/encode/vulkan_state_tracker.cpp]: Maybe not, the most recent push is now crashing on replay

**Review by mikes-lunarg — APPROVED** [review]
This change fixes a very real regression so I think it is good to merge. I want to understand why we can't treat the swapchain image lifetimes consistently between the state table and the handle table so I am going to keep looking at this, but it shouldn't hold up this PR.

**Review by panos-lunarg — APPROVED** [review]

**Pull request merged**